src/Ocelot/Configuration/File/FileGlobalConfiguration.cs

```diff
@@ -17,6 +17,8 @@ public FileGlobalConfiguration()
 
         public string RequestIdKey { get; set; }
 
+        public int? RequestTimeoutSeconds { get; set; }
```

Comment:
I would recommend using `int` instead of `int?` and set the default value (90 seconds) in the constructor. So, if not explicitely modified in the global configuration or for a given route, then 90 seconds will be used as default value. It is quite handy to do so, we avoid references to the default timeout value anywhere else.

Reply:
@ggnaegi Gui, do you realize the File model is solely a class for JSON de(serialization)? All File models must be devoid of business logic; otherwise, testing them would seem odd to me. However, I concur that a centralized location for maintaining a public default value is necessary. This could be a separate class or shared logic, which would then be utilized in `MessageInvokerPool` and Polly QoS classes. But assigning the value directly in the File model is not something I agree with.

> using `int` instead of `int?`

What is the issue with this alteration? Is it because it necessitates a review of current implementations? I have requested the author to adopt `Nullable<T>`. My rationale is to initiate a new design convention that mirrors the behavior of reference-type properties for value types, aiming to reduce redundant and confusing logic when a value is missing. Moreover, JSON (de)serialization relies on nulls! `Nullable<T>` is an excellent enhancement for reducing code, although it initially appears that we are writing more code for null checks. However, that's simply the nature of it.

> set the default value (90 seconds) in the constructor.

I must express my disagreement here. Constructors are invoked by JSON (de)serialization managers, and to my knowledge, the default constructor is what's called. We must exercise caution when defining an explicit default constructor. My recommendation is to avoid defining explicit default constructors whenever possible.

Reply:
@raman, ok, you can keep `int?` but the default value must be then set in the `ITimeoutCreator` implementation.

Reply:
Good!
Default 90 secs constant must be defined in `TimeoutCreator` class or in end consumers (old logic).

Do you agree to use `Nullable<T>` type more for data models? If Yes, it will be very good approved standard/convention in our project.

Reply:
> @raman, ok, you can keep `int?` but the default value must be then set in the `ITimeoutCreator` implementation.

No.

Reply:
@ggnaegi  Why are you calling phone numbers in India? 
### 🤣

Reply:
@raman Hello! It's a pleasure to meet you! Your username is outstanding. 😉